### PR TITLE
Enhance zypper package module error logs

### PR DESCRIPTION
**Pull request by Fdall**
The current package modules in cfengine hide all errors which make it quite bothersome to debug. Since the package module API offers a way to report errors, I did a small change to the zypper package module to make it reports if anything is prompted on stderr while calling the module.

It does not report on which item is causing the error since it is optional and would need way more changes in the module. Moreover, in most of the case, the error messages are explicit enough to find out the root issue.

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thanks for submitting a pull request! Maybe @craigcomstock can review this?

**Comment by cf-bottom** [comment]
Thanks for submitting a PR! Maybe @craigcomstock can review this?

**Comment by nickanderson** [comment]
Surfacing interactive prompts as errors seems reasonable to me.

**Comment by nickanderson** [comment]
https://tracker.mender.io/browse/CFE-3154

**Comment by nickanderson** [comment]
Can you please post example output before and after

**Comment by Fdall** [comment]
I attached a screen on the associated ticket. It seems that the hiding of the error does not come from the module nor the cfengine agent.

**Comment by nickanderson** [comment]
@olehermanse can you please take a look. This changes the behavior so that errors from the package manager are surfaced as an error in the cf-agent output instead of being emitted directly on stderr. See the screenshot in https://tracker.mender.io/browse/CFE-3154

File: modules/packages/zypper.in [hunk 1]
```diff
@@ -362,14 +371,14 @@ def repo_install():
...
+            subprocess_call(cmd_line, stdout=NULLFILE, stderr=subprocess.PIPE)
 
             # No final check. CFEngine will figure out that it's missing
             # if it failed.
 
-    # ret == 0 doesn't mean we succeeded with everything, but it's expensive to
+    # return 0 doesn't mean we succeeded with everything, but it's expensive to
     # check, so let CFEngine do that.
-    return ret
+    return 0
```
Comment (olehermanse) [inline modules/packages/zypper.in]: Why did you change these to always return 0 ? If that is really necessary, you have to explain why in the commit message.

File: modules/packages/zypper.in [hunk 2]
```diff
@@ -103,9 +103,18 @@ def subprocess_Popen(cmd, stdout=None, stderr=None):
     return result
 
 
+# If subprocess.PIPE is passed as stderr, it will re-write it in an ErrorMessage
 def subprocess_call(cmd, stdout=None, stderr=None):
     process = subprocess_Popen(cmd, stdout, stderr)
-    return process.wait()
+    return_code = process.wait()
```
Comment (vpodzime) [inline modules/packages/zypper.in]: This should use `process.communicate()` because otherwise the process may be stuck writing to its stderr while nothing is reading from it creating a deadlock here.

Comment (Fdall) [reply modules/packages/zypper.in]: Fixed in a new commit, I will rebase if needed.

**Comment by vpodzime** [comment]
Thanks for the fixup @Fdall! One last thing -- I guess SUSE is migrating to Python 3 just like other distros, right? So if we want to make this Python 3 compatible, we need to do `errs.decode("utf-8").split()` or even better `errs.decode("utf-8").splitlines()`.

**Comment by nickanderson** [comment]
@vpodzime opinions on picking to 3.12 and 3.10?
